### PR TITLE
Add API reference for on-demand jobs

### PR DESCRIPTION
**Pull request by magreenbaum**
Closes: https://github.com/meltano/cloud-docs/issues/7

Depends on merging of: https://github.com/meltano/infra/pull/273

### CODE REVIEW

File: docs/api_reference.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+API usage and examples for features of Meltano Cloud.
+
+The API functionalities documented here may be adjusted post-alpha phase.
+
+## On-demand Job Runners
+
+On-demand job runners allow users to rerun or run at any point a scheduled job.
+
+During the onboarding process, you will receive the following information that you will need to invoke your scheduled jobs on-demand:
+- An API key
```
Comment (WillDaSilva) [inline docs/api_reference.md:12]:
```suggestion
- An API key
```

File: docs/api_reference.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- The schedule name
+
+You will need to pass them in as headers (`x-api-key` and `meltano-runner-secret`) when invoking the API as part of the authentication process.
+
+### CURL Example
+
+First set `API_KEY` and `MELTANO_RUNNER_SECRET` environment variables locally.
+
+```
+curl -X POST "https://cloud-runners.meltano.com/v1/<tenant_resource_key>/<project_id>/<meltano_environment_name>/<schedule_name>" -H "x-api-key": ${API_KEY}" -H "meltano-runner-secret: ${MELTANO_RUNNER_SECRET}"
```
Comment (WillDaSilva) [inline docs/api_reference.md:28]: Maybe using env vars for each would be better?

```suggestion
curl -X POST "https://cloud-runners.meltano.com/v1/${TENANT_RESOURCE_KEY}/${PROJECT_ID}/${ENVIRONMENT_NAME}/${SCHEDULE_NAME}" -H "x-api-key": ${API_KEY}" -H "meltano-runner-secret: ${MELTANO_RUNNER_SECRET}"
```

File: docs/api_reference.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# API Reference
+
+API usage and examples for features of Meltano Cloud.
+
+The API functionalities documented here may be adjusted post-alpha phase.
```
Comment (WillDaSilva) [inline docs/api_reference.md:5]:
```suggestion
The API functionalities documented here may be adjusted during the Meltano Cloud alpha, and will definitely be adjusted prior to the Meltano Cloud GA release.
```

**Review by WillDaSilva — APPROVED** [review]
Looks good to me

File: docs/api_reference.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Information that you provide to us during [onboarding](https://github.com/meltano/cloud-docs/blob/main/docs/onboarding.md#step-1-submit-project-onboarding-information):
+- The name of the [Meltano Environment](https://docs.meltano.com/concepts/environments)
+- The schedule name
+
+You will need to pass them in as headers (`x-api-key` and `meltano-runner-secret`) when invoking the API as part of the authentication process.
+
+### CURL Example
+
+First set `API_KEY` and `MELTANO_RUNNER_SECRET` environment variables locally.
```
Comment (magreenbaum) [inline docs/api_reference.md:25]:
```suggestion
Set environment variables locally:
- `API_KEY`
- `MELTANO_RUNNER_SECRET`
- `TENANT_RESOURCE_KEY`
- `PROJECT_ID`
- `ENVIRONMENT_NAME`
- `SCHEDULE_NAME`
```

**Pull request merged**